runtime/interpreter/simplecompositevalue.go

```diff
@@ -39,6 +39,8 @@ type SimpleCompositeValue struct {
 	TypeID   sema.TypeID
 	// FieldNames are the names of the field members (i.e. not functions, and not computed fields), in order
 	FieldNames []string
+
+	isTransaction bool
```

Comment:
This is an unfortunate side effect. Needed a way to distinguish between transaction values and other composite values, because references cannot be created for transaction values, as they don't have a valid runtime type.
This doesn't need to be encoded/decoded, since transaction values doesn't get stored.

If there are any suggestions for a better way to do this, would love to hear.

Reply:
added https://github.com/onflow/cadence/pull/3235/commits/10649294381ac284df0625639c5f0f2694feba99